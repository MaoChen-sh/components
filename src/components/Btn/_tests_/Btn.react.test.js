import React from "react";
import Btn from "../Btn.js";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import * as TestUtils from "react-dom/test-utils";

describe("Btn", () => {
  const reactObj = <Btn>Btn Content</Btn>;
  it("renders correctly", () => {
    const tree = renderer.create(reactObj).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("mount correctly", () => {
    const btn = TestUtils.renderIntoDocument(reactObj);
    const dom = ReactDOM.findDOMNode(btn)
    expect(dom.textContent).toEqual("Btn Content");
  });
});
