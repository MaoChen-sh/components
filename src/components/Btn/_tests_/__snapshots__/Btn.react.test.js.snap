// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn renders correctly 1`] = `
<span
  className="sc-bdVaJa cSZdXv"
  onClick={[Function]}
>
  Btn Content
</span>
`;
